akka24-conductr-client-lib/src/main/scala/com/typesafe/conductr/clientlib/akka/ControlClient.scala

```diff
@@ -234,6 +306,67 @@ class ControlClient(handler: ConnectionHandler, conductrAddress: URL, apiVersion
     handler.withConnectedRequest(Payload.runBundle(bundleId, scale.getOrElse(DefaultScale), affinity))(handleAsHttpFailure[BundleRequestResult, BundleRequestSuccess, BundleRequestFailure])
 
   /**
+   * Scale a loaded bundle to a number of instances and returns a future.
+   * If `completeWhenScaleAchieved` is set to true, the future will be completed when the requested number of scale is
+   * achieved. Otherwise, the future is completed when the HTTP call to be control protocol is completed.
+   *
+   * @param bundleId An existing bundle identifier, a shortened version of it (min 7 characters) or
+   *                 a non-ambiguous name given to the bundle during loading.
+   * @param scale The number of instances of the bundle to start. Defaults to 1.
+   * @param affinity Optional: Identifier to other bundle.
+   *                 If specified, the current bundle will be run on the same host where
+   *                 the specified bundle is currently running.
+   * @param completeWhenScaleAchieved if set to true, then the future returned will be completed when bundle to be scaled reaches the
+   *                     number of requested instances.
+   *                     This is done by subscribing to bundle events SSE, and checking for number of running bundles
+   *                     whenever there are changes in the bundle events.
+   * @param cc implicit connection context
+   * @return The result as a Future[BundleRequestResult]. BundleRequestResult is a sealed trait and can be either:
+   *         - BundleRequestSuccess if the scaling request has been succeeded. This object contains the request and bundle id
+   *         - BundleRequestFailure if the scaling request has been failed. This object contains the HTTP status code and error message.
+   */
+  def runBundleComplete(bundleId: BundleId, scale: Option[Int] = None, affinity: Option[String] = None, completeWhenScaleAchieved: Boolean = true, completeTimeout: FiniteDuration = 30.seconds)(implicit cc: ConnectionContext): Future[BundleRequestResult] = {
```

Comment:
I had considered this before, but decided against it since if I were to do this, the existing `runBundle` will have a different behaviour altogether despite the same interface.

This is why I introduced `runBundleComplete` so the caller will need to consciously choose which behaviour is desirable.